tests/e2e/device-plugin_test.go

```diff
@@ -104,13 +125,16 @@ var _ = NVDescribe("GPU Device Plugin", func() {
 			Expect(err).NotTo(HaveOccurred())
 			_, err = helmClient.InstallChart(ctx, &chartSpec, nil)
 			Expect(err).NotTo(HaveOccurred())
+			diagnosticCollector, err = common.NewDiagnosticCollector(f.Namespace.Name, filepath.Join(*LogArtifactDir, f.UniqueName))
+			diagnosticCollector.ParseObjects(collectLogsFrom)
```

Comment:
If we always call `ParseObjects` here, does it make sense to use funcational arguments instead and rewrite this as:
```
common.NewDiagnosticCollector(
	common.WithNamespace(f.Namespace.Name), 
	common.WithArtifactDir(filepath.Join(*LogArtifactDir, f.UniqueName)),
	common.WithObjects(collectLogsFrom...),
)
```

Reply:
check